### ISSUE TITLE
Agetty service enabled for non-existent devices

### ISSUE THREAD

**Issue by raindev** [void-linux/void-mklive#193]
After the installation agetty is enabled for `/dev/hvsi0` (serial console?) and `/dev/hvc0` (Xen console) even if the devices are not present (which on many if not most of the systems):

```
$ ls /var/service/ | grep agetty
agetty-hvc0
agetty-hvsi0
agetty-tty1
agetty-tty2
agetty-tty3
agetty-tty4
agetty-tty5
agetty-tty6
```

While probably being necessary for installation in certain cases (I'm no expert here), the service fails if the console device is missing and is being constantly restarted. This results in in error messages like this appearing in the syslog every 10 seconds:

```
/dev/hvc0: cannot open as standard input: No such file or directory
```

**Comment by raindev** [comment]
It looks relatively straightforward to detect the actually existing devices [somewhere](https://github.com/void-linux/void-mklive/blob/aeabac5bacb6f9e15135d1d737fcdc4d15129b5a/installer.sh.in#L1328) in the installer and remove the services that do not have a corresponding `/dev/` entry.

**Comment by heliocat** [comment]
Another option would be to do something like:
```
tty=${PWD##*-}
[ -e "/dev/$tty" ] || pause
```
which would stop getty from launching against a tty that doesn't exist at all.

**Comment by raindev** [comment]
@heliocat, that will still result in the services starting and the processes to stay around, despite not doing anything, or am I missing something?

**Comment by heliocat** [comment]
It does. The problem is that there isn't a particularly great solution for this. In order of overall impact the options are: fake out the run script, set the service down, detect and delete the link out of /etc/runit/runsvdir/current. I can't remember the exact lifecycle of a terminal device so you may not want to simply delete the service symlink entirely if there's a chance that it becomes available later on in the life of the system since deleting things from the current set will affect all future boots.

**Comment by raindev** [comment]
Fixed in 51d4a5f4c2dad61e6b453b4f0e3dcd2b53304633 by including the services into the blacklist. I think it could be overwritten to enable the consoles if necessary, but it's not 100% clear to me how.
